libraries/conf_helpers.rb

```diff
@@ -57,5 +66,28 @@ def self.merge_config(current_config, desired_config)
       end
       stream.string
     end
+
+    class ConfContext
+      attr_reader :path
+      attr_reader :app
+      attr_accessor :stanza
+      attr_accessor :key
+
+      def initialize(path, stanza = nil, key = nil)
+        self.path = path
+        self.stanza = stanza
+        self.key = key
+      end
+
+      def path=(path)
+        @path = path
+        pathname = Pathname.new @path
+        @app = /local|default|metadata$/.match(pathname.parent.to_s) && pathname.parent.parent.basename.to_s
```

Comment:
Would the structure be more straightforward as: 
`@app = pathname.parent.parent.basename.to_s if regex.match(pathname.parent.to_s)`

Possibly premature regex optimization Should we also anchor the regex to a leading slash, and use a non-capturing group? e.g.  `[\/\\](?:local|default|metadata)$`